src/objectivec/test/ort_genai_api_test.mm

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import <XCTest/XCTest.h>
+
+#import "ort_genai_objc.h"
+#import "assertion_utils.h"
+#import <vector>
+#import <array>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OGATensor (Testing)
+
+- (nullable instancetype)initWithDataPointer:(void*)data
+                                       shape:(NSArray<NSNumber*>*)shape
+                                        type:(OGAElementType)elementType
+                                       error:(NSError**)error;
```

Comment:
instead of declaring `initWithDataPointer:shape:type:error:` again here, should we just make it part of OGATensor's public interface?

Reply:
In [C#](https://github.com/microsoft/onnxruntime-genai/blob/d9cb8548a576d0d5f43cb2214188268209cafd54/src/csharp/Tensor.cs#L38) and [Java](https://github.com/microsoft/onnxruntime-genai/blob/d9cb8548a576d0d5f43cb2214188268209cafd54/src/java/src/main/java/ai/onnxruntime/genai/Tensor.java#L75) bindings, the ctor of `OGATensor` is intentionally made `internal`. I think the reason is that it's not possible for users ton construct a native handle of `OGATensor`. I think in Objective-C I should follow the same fashion.

Reply:
ok, if there's a good reason to not make it public then we shouldn't. but as a follow up question, why is there a C API to create a tensor from a buffer?

https://github.com/microsoft/onnxruntime-genai/blob/4b12d57dfbc0097bf3f1b9c7660e2e79f02ff003/src/ort_genai_c.h#L440-L449

I wonder if the reason that Java and C# don't expose it is language-specific, e.g., because it is difficult to access native memory from a managed language?

Reply:
You're right. I was mistaken. There're **two** ctors in `OGATensor`. One is using raw buffer (`void*`) and another is using a `OgaTensor*`. This one `initWithDataPointer:shape:type:error:` is the raw buffer one and should made public, just like in C, C# and Java. The intentionally made internal one is the second one using a `OgaTensor*` as parameter, which is not possible to construct from outside this library.

Reply:
Fixed